src/responder/nss/nsssrv_mmap_cache.c

```diff
@@ -1445,8 +1446,16 @@ errno_t sss_mmap_cache_init(TALLOC_CTX *mem_ctx, const char *name,
     /* Attempt allocation several times, in case of EINTR */
     for (int i = 0; i < POSIX_FALLOCATE_ATTEMPTS; i++) {
         ret = posix_fallocate(mc_ctx->fd, 0, mc_ctx->mmap_size);
-        if (ret != EINTR)
-            break;
+        if (ret == EINTR) {
+            continue;
+        }
+        /* Copy-on-write file systems such as ZFS and Btrfs can't
+         * really support the posix_fallocate operation.
+         * Fall back to ftruncate() in this case */
+        if (ret == ENOSYS || ret == EOPNOTSUPP) {
+            ret = ftruncate(mc_ctx->fd, mc_ctx->mmap_size);
```

Comment:
Covscan complains here that `ftruncate()` may return '-1' and this is invalid value to provide to `strerror()` later.

I guess something like
```
if (ret == -1) {
    ret = errno;
}
```
is needed.